app/src/main/java/fr/neamar/kiss/loader/LoadContactsPojos.java

```diff
@@ -22,9 +27,26 @@ public LoadContactsPojos(Context context) {
         super(context, "contact://");
     }
 
+    private Pattern getMobileNumberPattern() {
+        InputStream inputStream = context.getResources().openRawResource(R.raw.phone_number_textable);
+        StringBuilder mobileDetectionRegex = new StringBuilder();
+
+        BufferedReader reader = new BufferedReader(new InputStreamReader(inputStream));
+
+        try {
+            String line;
+            while ((line = reader.readLine()) != null)
+                mobileDetectionRegex.append(line);
+
+            return Pattern.compile(mobileDetectionRegex.toString());
```

Comment:
Please only read and compile the regular expression only once, then cache it as a static class member (compilation could take 50-100ms total if you are unlucky).